src/Microsoft.Diagnostics.Runtime/src/DataReaders/MacOS/MacOSProcessDataReader.cs

```diff
@@ -84,11 +84,15 @@ public void Dispose()
             GC.SuppressFinalize(this);
         }
 
-        private void Dispose(bool _)
+        private void Dispose(bool disposing)
         {
+            _ = disposing;
             if (_disposed)
                 return;
 
+            foreach (uint threadAct in _threadActs.Values)
+                _ = Native.mach_port_deallocate(Native.mach_task_self(), threadAct);
```

Comment:
Hoist mach_task_self above the loop?

Reply:
Just came across this PR again, mach_task_self could actually be cached on the process reader instance.